src/parser/mod.rs

```diff
@@ -475,6 +475,10 @@ impl<'a> Parser<'a> {
                     if expecting_statement_delimiter && word.keyword == Keyword::END {
                         break;
                     }
+
+                    if expecting_statement_delimiter && word.keyword == Keyword::GO {
```

Comment:
Can we add a clarifying comment here, mentioning that for the `GO` statement, it is not necessary that the preceeding statement ends with a semicolon (ideally with a sql example)?

Reply:
Done 👍